webapp/src/components/shareBoard/shareBoard.tsx

```diff
@@ -264,18 +281,36 @@ export default function ShareBoardDialog(props: Props): JSX.Element {
         </span>
     )
 
-    const formatOptionLabel = (user: IUser) => {
+    const formatOptionLabel = (userOrChannel: IUser | Channel) => {
+        if ((userOrChannel as IUser).username) {
+            const user = userOrChannel as IUser
+            return(
+                <div className='user-item'>
+                    {Utils.isFocalboardPlugin() &&
+                        <img
+                            src={Utils.getProfilePicture(user.id)}
+                            className='user-item__img'
+                        />
+                    }
+                    <div className='ml-3'>
+                        <strong>{user.username}</strong>
+                        <strong className='ml-2 text-light'>{`@${user.username}`}</strong>
+                    </div>
+                </div>
+            )
+        }
+
+        if (!Utils.isFocalboardPlugin()) {
+            return null
+        }
+
+        const channel = userOrChannel as Channel
         return(
             <div className='user-item'>
-                {Utils.isFocalboardPlugin() &&
-                    <img
-                        src={Utils.getProfilePicture(user.id)}
-                        className='user-item__img'
-                    />
-                }
+                {channel.type === 'P' && <PrivateIcon/>}
+                {channel.type === 'O' && <PublicIcon/>}
```

Comment:
Constants already exist for these. https://github.com/mattermost/focalboard/blob/main/webapp/src/blocks/board.ts#L11